uc-crux-llvm/src/UCCrux/LLVM/Run/Check.hs

```diff
@@ -0,0 +1,406 @@
+                pure (findFun modCtx (FuncDefnSymbol func))
+              let argCTys = Crucible.cfgArgTypes ovCfg
+              ovFunCtx <-
+                case makeFunctionContext modCtx func argFTys argCTys of
+                  Left err ->
+                    panic
+                      "checkInferredContracts"
+                      [Text.unpack (ppFunctionContextError err)]
+                  Right funCtx' -> return funCtx'
+              let ?memOpts = CruxLLVM.memOpts llOpts
+              case testEquality argFTys types of
+                Nothing -> panic "checkInferredContracts" []
+                Just Refl ->
+                  do ref <- IORef.newIORef []
+                     return
+                       ( Sim.SymCreateOverrideFn $
+                           \_sym ->
+                             return $
+                               Check.createCheckOverride
+                                 appCtx
+                                 modCtx
+                                 ref
+                                 types
+                                 tcs
+                                 ovCfg
+                                 (FuncDefnSymbol func)
+                       , SomeCheckedCalls
+                           (\f -> f func ovFunCtx =<< IORef.readIORef ref)
+                       )
+        )
+
+-- | Postcondition: The keys of the returned 'Map' are exactly the
+-- 'EntryPoints'.
+checkInferredContractsFromEntryPoints ::
+  forall m arch msgs.
+  Crux.Logs msgs =>
+  Crux.SupportsCruxLogMessage msgs =>
+  ArchOk arch =>
+  AppContext ->
+  ModuleContext m arch ->
+  HandleAllocator ->
+  CruxOptions ->
+  LLVMOptions ->
+  -- | Where to begin symbolic execution
+  EntryPoints m ->
+  -- | Inferred function contracts
+  Map (DefnSymbol m) (Some (TypedConstraints m)) ->
+  IO (Map (DefnSymbol m) (SomeCheckResult m arch))
+checkInferredContractsFromEntryPoints appCtx modCtx halloc cruxOpts llOpts entries contracts =
+  fmap Map.fromList $
+    for (getEntryPoints entries) $
+      \entry ->
+        do CFGWithTypes cfg argFTys _retTy _varArgs <-
+             pure (findFun modCtx (FuncDefnSymbol entry))
+
+           funCtx <-
+             case makeFunctionContext modCtx entry argFTys (Crucible.cfgArgTypes cfg) of
+               Left err ->
+                 panic
+                   "checkInferredContracts"
+                   [Text.unpack (ppFunctionContextError err)]
+               Right funCtxF -> return funCtxF
+           result <-
+            checkInferredContracts
+              appCtx
+              modCtx
+              funCtx
+              halloc
+              cruxOpts
+              llOpts
+              (emptyConstraints argFTys)
+              cfg
+              contracts
+           return (entry, SomeCheckResult funCtx result)
+
+-- | Infer preconditions for a group of functions, then for the ones that are
+-- safe-with-preconditions, check their inferred preconditions by seeing if they
```

Comment:
Safe with preconditions, but also where no execution limits were hit

Reply:
You're right - fixed!